nav2_map_server/src/map_3d/pcl_helper.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2020 Shivam Pandey
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_map_server/map_3d/pcl_helper.hpp"
+
+#include <iostream>
+#include <vector>
+#include <string>
+
+namespace nav2_map_server
+{
+
+namespace map_3d
+{
+
+void modifyMsgFields(
+  sensor_msgs::msg::PointCloud2 & msg,
+  const std::vector<pcl::PCLPointField> & fields)
+{
+  msg.fields.clear();
+  for (auto & field : fields) {
+    sensor_msgs::msg::PointField new_field;
+    new_field.datatype = field.datatype;
+    new_field.name = field.name;
+    new_field.count = field.count;
+    new_field.offset = field.offset;
+    msg.fields.push_back(new_field);
+  }
+  std::cout << "[DEBUG] [pcl_helper]: Message field modification done" << std::endl;
+}
+
+void pclToMsg(
+  sensor_msgs::msg::PointCloud2 & msg,
+  const pcl::PCLPointCloud2::Ptr & cloud)
+{
+  msg.data.clear();
+  modifyMsgFields(msg, cloud->fields);
```

Comment:
I don't understand -- pcl_conversions has a msgs -> PC2 converter, why not use that? Actually, all of this file looks like it could be replaced with pcl_conversions functions